src/controller/jobRoleController.ts

```diff
@@ -5,16 +5,103 @@ import { bandService } from "../service/bandService";
 import { capabilityService } from "../service/capabilityService";
 import { validate } from "../validator/editJobRoleValidator";
 
+const appendURLParams = (params: URLSearchParams, name: string, object: any): void => {
+    if (object) {
+        if (Array.isArray(object)) {
+            params.append(name, encodeURIComponent(JSON.stringify(object)));
+        } else {
+            params.append(name, encodeURIComponent(object));
+        }
+    }
+}
+
+const encodeURLFilterParams = (name: string, band: string | string[], capability: string | string[]): string => {
+    const searchParams = new URLSearchParams();
+    appendURLParams(searchParams, "nameFilter", name);
+    appendURLParams(searchParams, "bandFilter", band);
+    appendURLParams(searchParams, "capabilityFilter", capability);
+    
+    return searchParams.toString();
+}
+
+const decodeURLFilterParams = (req: Request) => {
+    let filters = {} as any;
+    if (req.query?.nameFilter) {
+        filters.nameFilter = decodeURIComponent(req.query.nameFilter as string);
+    }
+    if (req.query?.bandFilter) {
+        const bandFilter = JSON.parse(decodeURIComponent(req.query.bandFilter as string));
+        let bandFilterArray: number[];
+        if (!Array.isArray(bandFilter)) {
+            bandFilterArray = [parseInt(bandFilter)];                    
+        } else {
+            bandFilterArray = bandFilter.map(band => parseInt(band));        
+        }
+        filters.bandFilter = bandFilterArray;
+    }
+    if (req.query?.capabilityFilter) {
+        const capabilityFilter = JSON.parse(decodeURIComponent(req.query.capabilityFilter as string));
+        let capabilityFilterArray: number[];
+        if (!Array.isArray(capabilityFilter)) {
+            capabilityFilterArray = [parseInt(capabilityFilter)];
+        } else {
+            capabilityFilterArray = capabilityFilter.map(capability => parseInt(capability));        
+        }
+        filters.capabilityFilter = capabilityFilterArray;
+    }
+    
+    return filters;
+}
+
```

Comment:
I think you should be able to split the content of the band and capability if statements into it's own function:

```
const filter = (filterString: string): [] number | undefined => {
   if(!filterString) {
      return undefiend
  }

   let filters = JSON.parse(decodeURIComponent(filterString)
    let filteredList: number[];
        if (!Array.isArray(filters)) {
            filteredList = [parseInt(filters)];
        } else {
            filteredList = filters.map(filter => parseInt(filter));        
        }
       return filteredList
}
```

Your code then becomes

```
const decodeURLFilterParams = (req: Request) => {
    let filters = {} as any;
    if (req.query?.nameFilter) {
        filters.nameFilter = decodeURIComponent(req.query.nameFilter as string);
    }

    filters.bandFilter = filter(req.query.bandFilter as string)
    filters.capabilityFilter = filter(req.query.capabilityFilter as string);

    return filters;
}
```